### DISCUSSION TITLE
Getting only one record instead of multiple records (Supabase functions)

### DISCUSSION THREAD

**Discussion by plxity** [supabase/supabase#5668] [Questions] [answered]
```
select * from user_data t1 where t1."userId" 
  not in (select match_request_to from matches_data where
  match_request_from in(uid)) 
```

I'm using this to fetch multiple records but getting only the first record.
Tried running in SQL editor, it is working fine there. Getting multiple records there.

**Comment by steve-chavez** [comment]
> I'm using this to fetch multiple records but getting only the first record.

How are you calling that query?

**Comment by plxity** [comment, marked as answer]
I'm using this - 
```
 const { data, error } = await supabase.rpc('fetch_test', {
      uid: '942674373424463873',
     });
```

**Reply by plxity** [reply]
And the return type is `record`

**Reply by GaryAustin1** [reply]
Did you test your function with sql editor or just your sql?
Anyway you would need  SETOF RECORD

**Reply by plxity** [reply]
![Screenshot 2022-02-26 at 2 31 05 AM](https://user-images.githubusercontent.com/28510494/155802418-7af4496f-fd0d-4f2b-af34-34060b6117cd.png)
I have this config. I have selected the language as SQL.

**Reply by GaryAustin1** [reply]
You'll need to write your function in the sql editor to do setof record return.  The function editor last I checked did not deal with arrays or setof as return items.
Edit: You will also have to drop the function first as you can't change return type (which is why you don't see it in the edit mode versus create mode) of the UI function editor.

**Reply by plxity** [reply]
![Screenshot 2022-02-26 at 2 55 47 AM](https://user-images.githubusercontent.com/28510494/155805095-d0162dec-8003-44b8-915f-d5320894cb33.png)
Is this config fine?

**Reply by GaryAustin1** [reply]
No.
You can't do this in the function editor.  You have to use the SQL Editor Window

It's going to be something like this:

```sql
drop function fetch_users();

create or replace function fetch_users(uid text)
returns setof record language sql  as $$
select * from user_data t1 where t1."userId" 
  not in (select match_request_to from matches_data where
  match_request_from in(uid));
$$;
```

But you use this window:
![image](https://user-images.githubusercontent.com/54564956/155807616-43be4610-acd8-422f-864d-2772a7d76884.png)

**Reply by plxity** [reply]
This worked. Thanks a ton, @GaryAustin1 :)

**Reply by one-dhiraj** [reply]
Thanks @GaryAustin1
